api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/v2/EligibilityAndEntitlementServiceV2.java

```diff
@@ -0,0 +1,85 @@
+package uk.gov.dhsc.htbhf.claimant.service.v2;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.dhsc.htbhf.claimant.entitlement.PaymentCycleEntitlementCalculator;
+import uk.gov.dhsc.htbhf.claimant.entitlement.PaymentCycleVoucherEntitlement;
+import uk.gov.dhsc.htbhf.claimant.entity.Claimant;
+import uk.gov.dhsc.htbhf.claimant.entity.PaymentCycle;
+import uk.gov.dhsc.htbhf.claimant.model.eligibility.EligibilityAndEntitlementDecision;
+import uk.gov.dhsc.htbhf.claimant.repository.ClaimRepository;
+import uk.gov.dhsc.htbhf.claimant.service.DuplicateClaimChecker;
+import uk.gov.dhsc.htbhf.claimant.service.EligibilityAndEntitlementDecisionFactory;
+import uk.gov.dhsc.htbhf.claimant.service.EligibilityAndEntitlementService;
+import uk.gov.dhsc.htbhf.dwp.model.v2.IdentityAndEligibilityResponse;
+
+import java.time.LocalDate;
+import java.util.Optional;
+import java.util.UUID;
+
+@AllArgsConstructor
+@Slf4j
```

Comment:
Needs to be a component. Can be done in this or next PR.

Reply:
No, it doesn't, there are now two implementations of the interface which means that we need to control which instance goes where. In the final PR for this story I will be using @Configuration beans to setup the v1 and v2 controllers and required services. Either way, not for this PR.

Reply:
I think we're going to have to wire these in as `@Bean`s and make V1 the primary.